### PR TITLE
Start adding validation tests

### PR DESCRIPTION
**Pull request by marot**
Add tests to verify that refactoring in issue https://asaphub.atlassian.net/browse/CRN-714 works

### CODE REVIEW

File: apps/asap-server/test/routes/user.route.test.ts [hunk 1]
```diff
@@ -95,6 +95,16 @@ describe('/users/ route', () => {
     });
 
     describe('Parameter validation', () => {
+      test('Should return a 400 error when additional properties exist', async () => {
+        const response = await supertest(appWithMockedAuth)
```
Comment (niezgoda) [inline apps/asap-server/test/routes/user.route.test.ts:99]: maybe you could use test.each?

Same code as on line 293-301

File: apps/asap-server/test/routes/news.route.test.ts [hunk 2]
```diff
@@ -53,6 +53,13 @@ describe('/news/ route', () => {
     });
 
     describe('Parameter validation', () => {
+      test('Should return a validation error when additional fields exist', async () => {
+        const response = await supertest(app).get(`/news`).query({
```
Comment (niezgoda) [inline apps/asap-server/test/routes/news.route.test.ts:57]: You can create new set of tests and put there all of the routers.
Then use test.each to validate invalid parameters.

**Review by niezgoda — CHANGES_REQUESTED** [review]

File: apps/asap-server/test/routes/user.route.test.ts (hunk 1 above)
Comment (marot) [reply apps/asap-server/test/routes/user.route.test.ts:99]: I understand the concern but in this case I'd prefer to be consistent with the existing code.
Let's discuss at a RTM if we should refactor authentication/validation tests and group them together.

File: apps/asap-server/test/routes/news.route.test.ts (hunk 2 above)
Comment (marot) [reply apps/asap-server/test/routes/news.route.test.ts:57]: see other thread

File: apps/asap-server/test/handlers/calendar/subscribe-handler.test.ts [hunk 3]
```diff
@@ -42,11 +42,39 @@ describe('Calendar handler', () => {
...
   afterEach(() => {
-    subscribe.mockClear();
-    unsubscribe.mockClear();
+    subscribe.mockReset();
+    unsubscribe.mockReset();
+    calendarControllerMock.update.mockReset();
   });
 
   describe('Validation', () => {
+    test.each([
```
Comment (peterstarling) [inline apps/asap-server/test/handlers/calendar/subscribe-handler.test.ts:51]: to me using `.each` like this (ie for only two cases and with such complicated setup in the array) seems unnecessary - it always makes it a little harder to read

**Review by peterstarling — APPROVED** [review]
looks good, just a minor comment + I would expect some description and a reference to a ticket?

**Review by niezgoda — APPROVED** [review]

File: apps/asap-server/test/handlers/calendar/subscribe-handler.test.ts (hunk 3 above)
Comment (marot) [reply apps/asap-server/test/handlers/calendar/subscribe-handler.test.ts:51]: @peterstarling yes I agree. I had more test cases before but deleted them after discussion with @kmaid. I'll remove the `.each`

**Pull request merged**